cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/KafkaSource.java

```diff
@@ -185,5 +241,36 @@ public String getKafkaBrokers() {
     public Long getDefaultOffset() {
       return defaultOffset;
     }
+
+    private FormatSpecification getFormatSpec() {
+      FormatSpecification formatSpec = null;
+      if (!Strings.isNullOrEmpty(format)) {
+        // try to parse the schema if there is one
+        Schema schemaObj = parseSchema();
+
+        // strip format.settings. from any properties and use them in the format spec
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
```

Comment:
Looks like builder only used in getProperties != null? You can move it inside the scope of the if-else